api/v1beta1/common_types.go

```diff
@@ -129,6 +129,14 @@ const (
 	// will receive the resulting object.
 	TopologyDryRunAnnotation = "topology.cluster.x-k8s.io/dry-run"
 
+	// OldDeploymentAnnotation is an annotation that gets set on MachineSet/Machine by the machinedeployment/machineset
+	// controller only during a rollout. It is used for tainting Nodes with PreferNoSchedule if the Nodes are in old
+	// MachineSets that will be deleted during a rollout. When a rollout happens, the machinedeployment controller should
+	// set this annotation to a MachineSet with an older revision. The machine controller then propagates the annotation
+	// from MachineSet to corresponding Machines and finally the machine controller taints Nodes that correspond to the
+	// annotated Machines.
+	OldDeploymentAnnotation = "cluster.x-k8s.io/old-deployment"
```

Comment:
I'd expect this to be something like `ToBeDeletedByClusterCAPI`
There's no old deployment there's decommissioned Machines owned by an old MachineSet belonging to a MachineDeployment

Reply:
wondering if we should revert the pattern and have something like cluster.x-k8s.io/current-revision on the current MachineSet only
@enxebre @sbueringer opinions?